WordPress/src/main/java/org/wordpress/android/widgets/WPNetworkImageView.java

```diff
@@ -183,6 +187,13 @@ private void loadImageIfNecessary(final boolean isInLayoutPass) {
             }
         }
 
+        // skip this URL if a previous request for it returned a 404
+        if (mUrlSkipList.contains(mUrl)) {
```

Comment:
I wonder If we should call `add` and `contains` in a synchronized way.
I see in the documentation of `Hashset` <blockquote>Note that this implementation is not synchronized. If multiple threads access a hash set concurrently, and at least one of the threads modifies the set, it must be synchronized externally.</blockquote>